main/src/main/java/com/airbnb/reair/incremental/db/PersistedJobInfoStore.java

```diff
@@ -411,6 +277,80 @@ public void run() throws Exception {
     }
   }
 
+  private synchronized void batchPersistNewImpl(List<PersistedJobInfo> jobs)
+      throws IOException, SQLException, StateUpdateException {
+    LOG.debug(String.format("Persisting %d PersistedJobInfos", jobs.size()));
+    if (jobs.size() == 0) {
+      return;
+    }
+    String query = generateQuery(jobs.size());
+    Connection connection = dbConnectionFactory.getConnection();
+    try (PreparedStatement ps =
+             connection.prepareStatement(query, Statement.RETURN_GENERATED_KEYS)) {
+      int queryParamIndex = 1;
+      for (PersistedJobInfo job: jobs) {
+        ps.setTimestamp(queryParamIndex++, new Timestamp(job.getCreateTime()));
+        ps.setString(queryParamIndex++, job.getOperation().toString());
+        ps.setString(queryParamIndex++, job.getStatus().toString());
+        ps.setString(queryParamIndex++, job.getSrcPath().map(Path::toString).orElse(null));
+        ps.setString(queryParamIndex++, job.getSrcClusterName());
+        ps.setString(queryParamIndex++, job.getSrcDbName());
+        ps.setString(queryParamIndex++, job.getSrcTableName());
+        ps.setString(queryParamIndex++, ReplicationUtils.convertToJson(job.getSrcPartitionNames()));
+        ps.setString(queryParamIndex++, job.getSrcObjectTldt().orElse(null));
+        ps.setString(queryParamIndex++, job.getRenameToDb().orElse(null));
+        ps.setString(queryParamIndex++, job.getRenameToTable().orElse(null));
+        ps.setString(queryParamIndex++, job.getRenameToPartition().orElse(null));
+        ps.setString(queryParamIndex++, job.getRenameToPath().map(Path::toString).orElse(null));
+        ps.setString(queryParamIndex++, ReplicationUtils.convertToJson(job.getExtras()));
+      }
+      ps.execute();
+      ResultSet rs = ps.getGeneratedKeys();
+      for (PersistedJobInfo j : jobs) {
+        rs.next();
+        j.setPersisted(rs.getLong(1));
+      }
+    }
+  }
+
+  /**
+   * Persists PENDING PersistedJobInfos to the DB.
+   * @param jobs a list of PersistedJobInfos in the PENDING state
+   * @throws StateUpdateException if there is a SQLException or any Infos are not PENDING
+   */
+  public synchronized void batchPersistNew(List<PersistedJobInfo> jobs)
```

Comment:
Nit: Name is a little odd. Why not `persist` or `persistMany` to follow the other method name that persists a single job?

Reply:
It should probably be batchCreate, or createMany, persistMany as a name doesn't make sense since the functionality is different than persist

Reply:
SGTM